docs/rhel4edge_iso.md

```diff
@@ -0,0 +1,212 @@
+# Install MicroShift on RHEL for Edge
+To install MicroShift in a production environment, it is necessary to create a RHEL for Edge ISO installer with all the necessary components preloaded on the image.
+
+## Build RHEL for Edge Installer ISO
+Log into the development virtual machine with the `microshift` user credentials.
+> The development machine configuration guidelines can be found in the [MicroShift Development Environment on RHEL 8.x](./devenv_rhel8.md) document.
+
+Follow the instructions in the [RPM Packages](./devenv_rhel8.md#rpm-packages) section to create MicroShift RPM packages.
+
+The scripts for building the installer are located in the `scripts/image-builder` subdirectory.
+
+### Prerequisites
+Execute the `scripts/image-builder/configure.sh` script to install the tools necessary for building the installer image.
+
+Download the OpenShift pull secret from the https://console.redhat.com/openshift/downloads#tool-pull-secret page and save it into the `~microshift/pull-secret.txt` file. 
+
+Make sure there is more than 20GB of free disk space necessary for the build artifacts. Run the following command to free the space if necessary.
+```bash
+./scripts/image-builder/cleanup.sh -full
```

Comment:
This step gave me an error

```
$ ./scripts/image-builder/cleanup.sh -full

# Cleaning the build directory

# Cleaning up local ostree container server
Untagged: k8s.gcr.io/pause:3.5
Untagged: quay.io/microshift/flannel-cni:v0.14.0
Untagged: quay.io/coreos/flannel:v0.14.0
Untagged: quay.io/kubevirt/hostpath-provisioner:v0.8.0
Untagged: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:94a4cd55cb067f6d3747843b5532114806b6c81e8421e370bfe92ccafcc6bca7
Untagged: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:4dcacb7ae5c5a5c0cb2fbc8e132b14a72ee937b3dd7d201c7a66ded05e5d6790
Untagged: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:608a8c9f893b3fae358028b44238f3184af2f3bd5cde844b8b2688583b8468ea
Untagged: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6e963275041b08eafa47fa119287e1d4428b5af642c3befc6f0c90900d3516a4
Untagged: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:08e8b4004edaeeb125ced09ab2c4cd6d690afaf3a86309c91a994dec8e3ccbf3
Deleted: ed210e3e4a5bae1237f1bb44d72a05a2f1e5c6bfe7a7e73da179e2534269c459
Deleted: 4324dc7a1ffa5b107debea9236b2320af76269d6d24fd49dc7c37ddb8d26b75d
Deleted: 8522d622299ca431311ac69992419c956fbaca6fa8289c76810c9399d17c69de
Deleted: 7cbc61ff04c899eb7474316691aa567604d280e36d25acf8dda85d9bd7104ae3
Deleted: b11cde7e90ec607373e739dbc7e747ac049eacfa66ef97e1dc02e58219d83f11
Deleted: 7b5d2cc656439ec9bc9654639f13120382e3bc0698b5212f85ec2b23723732fa
Deleted: cfed419e1e2e4ba7724ba31a1e03134b1f32a4f08a9f16fcbecb72a54bf2aedf
Deleted: caac97eafff18509656533ff2b681199c7cd868ab526233321ae4b6eeeb5901f
Deleted: 55e294662f2c60c72271f4261f19b1be9c52515ffc76ba79f93bb44022fec181
Error: removing image ecbf84971ab68c43eb1d30c9b85a079f51821c63f923fc6d54ed6fd3257997be: container 2819a57202965d68b80dc580125bebe8ccd805fbf6953baeede5643ea127a3ca using image could not be removed: error freeing lock for container 2819a57202965d68b80dc580125bebe8ccd805fbf6953baeede5643ea127a3ca: no such file or directory
```

Running it again let it complete, so I wonder if there's a step in the cleanup script that doesn't wait long enough?

```
$ ./scripts/image-builder/cleanup.sh -full

# Cleaning the build directory

# Cleaning up local ostree container server
Untagged: localhost/ostree-demo:0.0.3
Deleted: ecbf84971ab68c43eb1d30c9b85a079f51821c63f923fc6d54ed6fd3257997be

# Cancelling composer jobs

# Deleting composer jobs

# Cleaning up composer sources

# Clean up user cache

# Clean osbuild worker cache
Warning: Stopping osbuild-composer.service, but it can still be activated by:
  osbuild-local-worker.socket
```

Reply:
Added a wait loop in the script to make sure microshift process ends before attempt to clean up images.